### DISCUSSION TITLE
How to deal with Time History of Unsteady-State Fluid Dynamics while using DeepXDE during the Training?

### DISCUSSION THREAD

**Discussion by Frankchu1999** [lululxvi/deepxde#1885] [Q&A]
Hello,

I am researching the use of Physics-Informed Neural Networks (PINNs) to predict the force output from a flipper on a sea turtle-inspired robot, which involves unsteady-state fluid dynamics. I noticed the heat transfer example in DeepXDE, which has similar boundary and initial conditions to fluid dynamics problems. However, I am particularly interested in understanding how to handle **Time History** during training for unsteady-state fluid dynamics.

In PINNs, t is often treated as an input to approximate the solution  𝑢(𝑥,𝑦,𝑡). However, this doesn’t inherently capture time history, where the solution at time 𝑡 depends on earlier states. How can DeepXDE handle this causality issue? Are techniques like iterative training, recurrent architectures, or loss constraints recommended?

Could anyone provide insights or examples on how DeepXDE manages time-dependent aspects in training for unsteady fluid flows? Specifically, I’d like to know how to set up the training process to account for the time evolution in the data.

Thank you very much for your help!

**Comment by lululxvi** [comment]
In many cases, you don't need this. If you really want to use this, it is easy to do and there are many papers on this topic.
